pkg/api/api.go

```diff
@@ -354,12 +363,24 @@ func (a *API) RegisterStoreGateway(s *storegateway.StoreGateway) {
 	a.RegisterRoute("/store-gateway/tenant/{tenant}/blocks", http.HandlerFunc(s.BlocksHandler), false, true, "GET")
 }
 
-// RegisterCompactor registers the ring UI page associated with the compactor.
+// RegisterCompactor registers routes associated with the compactor.
 func (a *API) RegisterCompactor(c *compactor.MultitenantCompactor) {
 	a.indexPage.AddLinks(defaultWeight, "Compactor", []IndexPageLink{
 		{Desc: "Ring status", Path: "/compactor/ring"},
 	})
 	a.RegisterRoute("/compactor/ring", http.HandlerFunc(c.RingHandler), false, true, "GET", "POST")
+	// Endpoint to handle requests for completion of block backfilling.
+	// Placed before less specific handler for same route, that doesn't take query parameter, for precedence.
+	a.RegisterRouteWithQueryParameters("/api/v1/upload/block/{block}", http.HandlerFunc(c.CompleteBlockUpload),
```

Comment:
I would recommend against different handlers handling same route but differentiating on query parameter. It's confusing, and requires extra code to setup. Doing `if` in the handler, or using different route, look like easier options.

Reply:
@pstibrany I was originally using a different route, but went with @pracucci's suggestion to use a query parameter. What do you think is preferable, keeping the query parameter and using an `if` in the handler?

Reply:
Doing `RegisterRoute` and using `if` in the handler.